### ISSUE TITLE
Ember server appears to hang at shutdown on CE 3.5.0-RC1

### ISSUE THREAD

**Issue by kubukoz** [http4s/http4s#6978]
Perhaps this has been reported before either here or in cats-effect, but here it is.

This runs perfectly fine, exits almost immediately:

```scala
//> using scala "2.13.10"
//> using lib "org.typelevel::cats-effect:3.4.7"
//> using lib "org.http4s::http4s-ember-server:0.23.18"
import cats.effect.IOApp
import org.http4s.ember.server.EmberServerBuilder
import cats.effect.IO

object main extends IOApp.Simple {
  def run = EmberServerBuilder.default[IO].build.use_
}
```

However, upgrade CE to 3.5.0-RC1 and it never exits, and also can't be killed with SIGTERM.

This also happens in `3.5-204fca2` (latest CE as of https://github.com/typelevel/cats-effect/tree/204fca22af47e48c43fa72913a2b108345ab2e74) and with a fresh http4s snapshot from https://github.com/http4s/http4s/tree/bdd0192580311558d69727723a6d974b7e561e07 .

**Comment by kubukoz** [comment]
fyi @djspiewak @armanbilge

**Comment by armanbilge** [comment]
Thanks, it needs the fixes in this PR.
- https://github.com/typelevel/fs2/pull/3142

**Comment by kubukoz** [comment]
glad you have it on the radar, then :)

**Comment by armanbilge** [comment]
@kubukoz I'm more worried about blaze fwiw
- https://github.com/http4s/blaze/issues/772

**Comment by djspiewak** [comment]
We've also found some issues in the timer state machine. Definitely going to need an RC2. I'm working through all that now.

**Comment by armanbilge** [comment]
https://github.com/typelevel/fs2/releases/tag/v3.7.0-RC1

**Comment by armanbilge** [comment]
Just confirmed this is fixed in http4s `0.23.19-RC1`.
